src/compute/curve.rs

```diff
@@ -1,24 +1,34 @@
 use crate::compute::ElementP2;
 use curve25519_dalek::ristretto::RistrettoPoint;
 
-pub trait Curve {
+pub trait SwCurveConfig: ark_ec::short_weierstrass::SWCurveConfig {
     fn curve_id() -> u32;
 }
 
-impl Curve for RistrettoPoint {
+impl SwCurveConfig for ark_bls12_381::g1::Config {
     fn curve_id() -> u32 {
-        blitzar_sys::SXT_CURVE_RISTRETTO255
+        blitzar_sys::SXT_CURVE_BLS_381
     }
 }
 
-impl Curve for ElementP2<ark_bls12_381::g1::Config> {
+impl SwCurveConfig for ark_bn254::g1::Config {
     fn curve_id() -> u32 {
-        blitzar_sys::SXT_CURVE_BLS_381
+        blitzar_sys::SXT_CURVE_BN_254
     }
 }
 
-impl Curve for ElementP2<ark_bn254::g1::Config> {
+pub trait Curve {
+    fn curve_id() -> u32;
+}
+
+impl Curve for RistrettoPoint {
     fn curve_id() -> u32 {
-        blitzar_sys::SXT_CURVE_BN_254
+        blitzar_sys::SXT_CURVE_RISTRETTO255
+    }
+}
+
+impl<C: SwCurveConfig> Curve for ElementP2<C> {
```

Comment:
Doing this would remove the need for the new `SwCurveConfig` trait and would let you `impl Curve for ark_bls12_381::g1::Config`. It's not really a `Curve`, but you could rename the trait to `CurveId` if you don't like the `Curve` name at that point.
```suggestion
impl<C: ark_ec::short_weierstrass::SWCurveConfig + Curve> Curve for ElementP2<C> {
```

Reply:
I changed to an associated constant. I kept SwCurveConfig since it's used to constrain the extension trait SwMsmHandle.

Reply:
You could do this to get around that:
`impl<C: ark_ec::short_weierstrass::SWCurveConfig + CurveId + Clone> SwMsmHandle for MsmHandle<ElementP2<C>>`